### DISCUSSION TITLE
How to setup to Keep the cursor at the bottom line of selected text after yank it

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [JetBrains/ideavim#1030] [Q&A] [answered]
Hi,

In LazyVim yanki provides that when you select a multiline text and press `y` the cursor remains at the bottom line of the selected text or it gets into the following line, I don't remember exactly.
While the default behaviour of vim/neovim is that the cursor goes back to the top line of the selected text.
How can I replicate this behaviour in IdeaVim? Is there a way to include yanki?

**Comment by jphalip** [comment, marked as answer]
In case it helps, this is what I use to at least keep the cursor it is position:

```
vnoremap y m`y``
```

This works in regular Vim too, not just IdeaVim.

**Reply by Andras-Csanyi** [reply]
You are my best friend! Thank you!

**Reply by jphalip** [reply]
Actually, I just realize this creates an issue, as with this you can't save the selected text to a specific register. Try for example: `V4j"ay`.

In Vim, this should work instead:

```
vnoremap <expr> y 'm`"' . v:register . 'y``'
```

However, this isn't supported in IdeaVim yet. I've started trying to implement it but then ran into an issue, which I've just posted about here: https://github.com/JetBrains/ideavim/discussions/1045

I'll update this discussion if I find a solution.
